scripts/utils/pre-release.mjs

```diff
@@ -0,0 +1,370 @@
+import moment from 'moment';
+import replace from 'replace-in-file';
+import inquirer from 'inquirer';
+import semver from 'semver';
+import {
+  displayErrorMessage,
+  displayConfirmationMessage
+} from './index.mjs';
+import JSDOMGlobal from 'jsdom-global';
+import chalk from 'chalk';
+import glob from 'glob';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+// Using `require` exclusively to import `json` files. Prevents 'experimental' warnings from being thrown in the
+// console.
+const require = createRequire(import.meta.url);
+const hotPackageJson = require('../../package.json');
+const workspacePackages = hotPackageJson.workspaces.packages;
+
+/**
+ * Check if the provided version number is a valid semver version number.
+ *
+ * @param {string} version Version number.
+ * @returns {boolean} `true` if the version number is a valid semver version number, `false` otherwise.
+ */
+export function isVersionValid(version) {
+  return !!semver.valid(version);
+}
+
+/**
+ * Check if the provided release date in a format of 'DD/MM/YYYY' is a valid future date.
+ *
+ * @param {string} date The date in format of 'DD/MM/YYYY'.
+ * @returns {object} Object containing information about the release validity in a form of `{valid: boolean, error:
+ * string}`.
+ */
+export function validateReleaseDate(date) {
+  const dateObj = moment(date, 'DD/MM/YYYY');
+  const now = moment();
+  const returnObj = {
+    valid: true,
+    error: null
+  };
+
+  if (!dateObj.isValid()) {
+    returnObj.valid = false;
+    returnObj.error = 'The provided date is invalid.';
+
+  } else if (!dateObj.isAfter(now)) {
+    returnObj.valid = false;
+    returnObj.error = 'The release date has to be a future date.';
+  }
+
+  return returnObj;
+}
+
+/**
+ * Set the provided version number to the Handsontable's `package.json` and other packages' `dependency` fields.
+ *
+ * @param {string} version The version number.
+ * @param {Array} [packages] Array of package paths. Defaults to the workspace config.
+ */
+export function setVersion(version, packages = workspacePackages) {
+  let versionReplaced = true;
+
+  packages.forEach((packagesLocation) => {
+    const replacementStatus = replace.sync({
+      files: `${packagesLocation}${packagesLocation === '.' ? '' : '*'}/package.json`,
+      from: [/"version": "(.*)"/, /"handsontable": "([^\d]*)((\d+)\.(\d+).(\d+)(.*))"/g],
+      to: (fullMatch, ...args) => {
+        if (fullMatch.indexOf('version') > 0) {
+          // Replace the version with the new version.
+          return `"version": "${version}"`;
+
+        } else {
+          const prerelease = !!semver.prerelease(version);
+          const depVersion = `${args[0]}${parseInt(semver.major(version), 10) - (+prerelease)}.0.0`;
+
+          // Replace the `handsontable` dependency with the current major (or previous major, if it's a prerelease).
+          return `"handsontable": "${depVersion}"`;
+        }
+      },
```

Comment:
I've found that SemVer `maxSatisfying` method can simplify the code 
```suggestion
      to: (fullMatch, ...[semverPrefix, previousVersion]) => {
        if (fullMatch.indexOf('version') > 0) {
          // Replace the version with the new version.
          return `"version": "${version}"`;

        } else {
          const maxSatisfyingVersion = semver.maxSatisfying([version, previousVersion], '*');

          // Replace the `handsontable` dependency with the current major (or previous major, if it's a prerelease).
          return `"handsontable": "${semverPrefix}${maxSatisfyingVersion}"`;
        }
      },
```

Reply:
I don't think it would work properly.
If:
```
const previousVersion = '3.4.5';
const version = '4.0.0-alpha.1';
```
then it would change the `handsontable` version for the dependencies to `${prefix}3.4.5`, as 
```
semver.maxSatisfying(['4.0.0-alpha.1', '3.4.5'], '*')
// 3.4.5
```
and we're trying to change this to `${prefix}3.0.0`.

Reply:
Thanks for noticing this. Then I can propose using the `major` method to extract the major number from the version. Something like this:
```js
const maxSatisfyingVersion = semver.major(semver.maxSatisfying([version, previousVersion], '*')) + '0.0';
```
Does this do the job?